person.rb

```diff
@@ -1,15 +1,21 @@
+require './nameable'
 # rubocop:disable Naming/PredicateName
 
-class Person
+class Person < Nameable
+  attr_reader :id
+  attr_accessor :name, :age, :parent_permission
+
   def initialize(age, name = 'unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    super()
   end
 
-  attr_reader :id
-  attr_accessor :name, :age, :parent_permission
+  def correct_name
+    @name.correct_name
```

Comment:
- [ ] Since this is supposed to be the concrete component that provides default implementation, you should return the name as it is so it can be decorated. This needs to return the @name so the decorators can use the name property and modify it by capitalizing or trimming the return from this method :)
``` ruby
  def correct_name
    @name
  end
```